src/backend/distributed/executor/executor.c

```diff
@@ -1646,8 +1643,9 @@ TransactionStateMachine(WorkerSession *session)
 					transaction->transactionState = REMOTE_TRANS_SENT_COMMAND;
```

Comment:
Not changed by this PR, but I think we should be setting the wait flags to WL_SOCKET_READABLE above when there are no tasks to only check for notices.

Reply:
Shall I add a commit to this PR?

Reply:
yes